src/core/curry.js

```diff
@@ -5,12 +5,15 @@ const isFunction = require('./isFunction')
 
 function applyCurry(fn, arg) {
   if(!isFunction(fn)) { return fn }
+
   return fn.length > 1 ? fn.bind(null, arg) : fn.call(null, arg)
 }
 
 // curry : ((a, b, c) -> d) -> a -> b -> c -> d
 function curry(fn) {
-  return function(...xs) {
+  if(fn.isCurried) { return fn }
+
+  const curried = function(...xs) {
```

Comment:
Could this become a function as opposed to assignment.

Reply:
Not sure what you mean by ' function as opposed to assignment', do you mean `function curried(...xs)` ?

Reply:
yeppers, named function I should of said.